backend/src/main/scala/cromwell/backend/package.scala

```diff
@@ -7,16 +7,27 @@ import wdl4s.values.WdlValue
 
 package object backend {
 
+  trait JobKey {
+    def scope: Scope
```

Comment:
It seems the reason why it's a `Scope` is because there's a [`ScatterKey`](https://github.com/broadinstitute/cromwell/blob/494aa86713ee3a19c3d944355672862efd292e3f/engine/src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala#L35)

The only reason why I (slightly) prefer `JobKey` over `ScopeKey` is that implementers of this trait typically use the word "job" to describe themselves.  But I can see good arguments for using `ScopeKey`.  Anybody else have a preference?